### PR TITLE
Order payment status performance

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it improves performance of order paymentStatus and paymentStatusDisplay resolving.

Related tickets:
[SALEOR-2737](https://app.clickup.com/t/2549495/SALEOR-2737)
[SALEOR-2738](https://app.clickup.com/t/2549495/SALEOR-2738)

Local testing (10 orders, 3 payments for each order):
before: "query-count": 36, "duplicates": 10
after: "query-count": 17, "duplicates": 0

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [X] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2737 Optimize paymentStatus resolver in Order type](https://app.clickup.com/t/2549495/SALEOR-2737)

**Comment by patrys** [comment]
Task linked: [SALEOR-2738 Optimize paymentStatusDisplay resolver in Order type](https://app.clickup.com/t/2549495/SALEOR-2738)

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/order/dataloaders.py [hunk 1]
```diff
@@ -51,6 +52,17 @@ def batch_load(self, keys):
         return [line_map.get(order_id, []) for order_id in keys]
 
 
+class PaymentsByOrderIdLoader(DataLoader):
+    context_key = "payments_by_order"
+
+    def batch_load(self, keys):
+        payments = Payment.objects.filter(order_id__in=keys).order_by("pk")
+        payment_map = defaultdict(list)
+        for line in payments.iterator():
```
Comment (korycins) [inline saleor/graphql/order/dataloaders.py:61]: Why `line` ?

Comment (tomaszszymanski129) [reply saleor/graphql/order/dataloaders.py:61]: copy pasted it a bit too much, thanks for the catch!

File: saleor/graphql/order/types.py [hunk 2]
```diff
@@ -817,12 +820,30 @@ def resolve_number(root: models.Order, _info):
         return str(root.pk)
 
     @staticmethod
-    def resolve_payment_status(root: models.Order, _info):
-        return root.get_payment_status()
```
Comment (fowczarek) [inline saleor/graphql/order/types.py:821]: Should we drop `get_payment_status`?

File: saleor/graphql/order/types.py [hunk 3]
```diff
@@ -817,12 +820,30 @@ def resolve_number(root: models.Order, _info):
...
+
+        return (
+            PaymentsByOrderIdLoader(info.context)
+            .load(root.id)
+            .then(_resolve_payment_status)
+        )
 
     @staticmethod
-    def resolve_payment_status_display(root: models.Order, _info):
-        return root.get_payment_status_display()
```
Comment (fowczarek) [inline saleor/graphql/order/types.py:825]: Should we drop `get_payment_status_display`?

**Review by korycins — APPROVED** [review]

File: saleor/graphql/order/types.py (hunk 2 above)
Comment (tomaszszymanski129) [reply saleor/graphql/order/types.py:821]: I think so. It's not used anywhere now so let's get rid of it.

**Review by fowczarek — APPROVED** [review]

**Pull request merged**